.github/workflows/datadog-go.yaml

```diff
@@ -0,0 +1,25 @@
+# We use github actions to test the code on windows and linux amd64. Circleci is used for linux arm64.
+
+name: datadog-go_windows
+on:
```

Comment:
I'd also suggest you to use the `pull_request`  trigger which is pretty useful as it runs on the merge result of the PR, while `push` is on the commits pushed to the branch (so pull_request ends up being more useful than the push one in my opinion)